DataFormats/PatCandidates/src/classes_def_objects.xml

```diff
@@ -334,7 +334,8 @@
   <class name="pat::PackedCandidate::PackedCovariance" ClassVersion="3">
    <version ClassVersion="3" checksum="3320406063"/>
   </class>
-  <class name="pat::PackedCandidate" ClassVersion="33">
+  <class name="pat::PackedCandidate" ClassVersion="34">
+   <version ClassVersion="34" checksum="1793021019"/>
```

Comment:
Based on the discussion in this PR and past available implementations of the PATPackedCandidateProducer,
please consider to add an ioread rule ``` if(rawCaloFraction_ != 0) 	rawHcalFraction_ = hcalFraction_ ;```.
This way rawHcalFraction() will return the same value for old and new data.
This is more of a magic for support of different ways how values were filled rather than something specific to the data format change.

Reply:
Implemented as follows:
@peruzzim suggested to fill the raw HCAL fraction only for charged particles (since this is the subset of candidates for which the raw HCAL Fraction has been saved into the hcalFraction_ member variable), and he also suggested to set the HCAL fraction to 0 for charged particles such that the new interface is fully consistent